File: app/src/main/java/com/zhouyu/android_create/fragments/BaseFragment.java
package com.zhouyu.android_create.fragments;

import android.app.Activity;
import android.view.View;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;

public class BaseFragment extends Fragment {
    public View view;
    public final <T extends View> T findViewById(int id) {
        return view.findViewById(id);
    }
    public void runUiThread(Runnable runnable) {
        FragmentActivity activity = getActivity();
        if(activity != null){
            activity.runOnUiThread(runnable);
        }
    }

    public Runnable executeThread(Runnable runnable){
        try {
            new Thread(runnable).start();
        }catch (Exception e){
            e.printStackTrace();
        }
        return runnable;
    }
}

File: app/src/main/java/com/zhouyu/pet_science/fragments/BaseFragment.kt
package com.zhouyu.pet_science.fragments

import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.zhouyu.pet_science.utils.PhoneMessage

open class BaseFragment : Fragment() {
    private var layoutView: View? = null
    fun <T : View?> findViewById(id: Int): T {
        return layoutView!!.findViewById(id)
    }
    fun runUiThread(runnable: Runnable?) {
        if(activity == null || activity?.isFinishing == true || activity?.isDestroyed == true){
            return
        }
        if(isDetached){
            return
        }
        activity?.runOnUiThread(runnable)
    }

    fun executeThread(runnable: Runnable): Runnable {
        try {
            Thread(runnable).start()
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return runnable
    }

    fun setTopBarView(view: View, isPadding: Boolean) {
        if (isPadding) {
            view.setPadding(
                view.paddingLeft,
                PhoneMessage.statusBarHeight,
                view.paddingRight,
                view.paddingBottom
            )
        } else {
            val layoutParams = view.layoutParams
            if (layoutParams is ViewGroup.MarginLayoutParams) {
                layoutParams.topMargin = PhoneMessage.statusBarHeight
                view.layoutParams = layoutParams
            }
        }
    }

    /**
     * 检查Fragment是否可见
     */
    fun isFragmentVisible(): Boolean {
        return isAdded && isVisible && userVisibleHint
    }
}
